### ISSUE TITLE
FileNotFoundException app_amplitude-disk-queue  Crash on play Store report

### ISSUE THREAD

**Issue by JulienGenoud** [amplitude/Amplitude-Kotlin#43]
I Assume it's from the new SDK but i'm getting this crash on one playstore report : 

`java.io.FileNotFoundException: /data/user/0/app.arcopypaste/app_amplitude-disk-queue/a56334788719a29ef5f5163bc266c180-4 (No such file or directory)`

Stacktrace : 
```
FATAL EXCEPTION: pool-14-thread-1
Process: app.arcopypaste, PID: 24792
java.io.FileNotFoundException: /data/user/0/app.arcopypaste/app_amplitude-disk-queue/a56334788719a29ef5f5163bc266c180-4 (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:200)
	at java.io.FileInputStream.<init>(FileInputStream.java:150)
	at v4.b.a(Unknown Source:16)
	at a5.d.n(SourceFile:13)
	at me.a.l(Unknown Source:8)
	at cf.k0.run(Unknown Source:86)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)
```

Device details : 

> Model name Redmi 6A
> Manufacturer xiaomi
> Android version Android 8.1 (SDK 27)
> Locale hi
> Screen size 720 x 1440

I'm rolling back to the old SDK.

**Comment by qingzhuozhen** [comment]
Hi, @JulienGenoud Thanks for reporting. We will look into this. cc @liuyang1520

**Comment by tanishq14developer** [comment]
Getting same issue

**Comment by bohan-amplitude** [comment]
Hi @tanishq14developer @JulienGenoud , are there more exception detail for this error? I'm working on catching the file not found exception and handle them for android storage but want to make sure I got the right place and root cause.

**Comment by JulienGenoud** [comment]
Hi @bohan-amplitude sure i've got this one as well : 
```
FATAL EXCEPTION: pool-15-thread-1
Process: app.arcopypaste, PID: 5967
java.io.FileNotFoundException: /data/user/0/app.arcopypaste/app_amplitude-disk-queue/a56334788719a29ef5f5163bc266c180-0: open failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.open(IoBridge.java:496)
	at java.io.FileInputStream.<init>(FileInputStream.java:159)
	at l4.b.a(Unknown Source:16)
	at q4.d.l(SourceFile:13)
	at ee.a.o(Unknown Source:8)
	at te.k0.run(Unknown Source:86)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
	Suppressed: c4.c: [q1{Cancelling}@dae4a36, java.util.concurrent.Executors$FinalizableDelegatedExecutorService@9963d37]
Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
	at libcore.io.Linux.open(Native Method)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
	at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7893)
	at libcore.io.IoBridge.open(IoBridge.java:482)
	... 8 more
```


and attached you the full logcat from the play store (idk if it helps) : 

[logcat (2).txt](https://github.com/amplitude/Amplitude-Kotlin/files/9157464/logcat.2.txt)

**Comment by tanishq14developer** [comment]
Fatal Exception: java.io.FileNotFoundException: /data/user/0/com.examlounge.examloungeapp/app_amplitude-disk-queue/08c06714d9b348ba23c1ad6e1dbc0a32-5: open failed: ENOENT (No such file or directory)
       at libcore.io.IoBridge.open(IoBridge.java:497)
       at java.io.FileInputStream.<init>(FileInputStream.java:159)
       at com.amplitude.android.utilities.AndroidStorage.getEventsString(AndroidStorage.java:16)
       at com.amplitude.core.platform.EventPipeline$upload$1.invokeSuspend(EventPipeline.java:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:8)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:90)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)

**Comment by tanishq14developer** [comment]
Caused by android.system.ErrnoException: open failed: ENOENT (No such file or directory)
       at libcore.io.Linux.open(Linux.java)
       at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
       at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
       at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
       at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7461)
       at libcore.io.IoBridge.open(IoBridge.java:483)
       at java.io.FileInputStream.<init>(FileInputStream.java:159)
       at com.amplitude.android.utilities.AndroidStorage.getEventsString(AndroidStorage.java:16)
       at com.amplitude.core.platform.EventPipeline$upload$1.invokeSuspend(EventPipeline.java:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:8)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:90)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)

**Comment by tanishq14developer** [comment]
Please fix this we have heavy audience and our Crashlytics is showing this error!! Please fix this as soon as possible

**Comment by bohan-amplitude** [comment]
@tanishq14developer @JulienGenoud Thanks for providing logs. Very helpful for me to debug. I released the v1.0.1 version to handle the FileNotFoundException.
